### DISCUSSION TITLE
"Connection to X-Touch Mini not possible"... Please help...

### DISCUSSION THREAD

**Discussion by GrimCultZero** [maartentamboer/X-Touch-Mini-FS2020#124] [Q&A]
Guys, please help. I keep getting this message and nothing is working...
![image](https://user-images.githubusercontent.com/126724797/223846664-2e9103e3-0f24-4f8d-8ca9-5357e5cb40d4.png)

**Comment by Koseng** [comment]
You need to adjust your configuration in config.json. The numbers can be different on different systems.

Check what is printed in the console after "Midi Input Devices" and "Midi Output Devices" and what number is behind X-TOUCH MINI.

Then adjust in the config file:

config.json
"midi_input_device": "X-TOUCH MINI **0**",
"midi_output_device": "X-TOUCH MINI **1**"

**Reply by GrimCultZero** [reply]
Thank you, @Koseng for your reply.
I did that already, and it's not helping. Here is how it looks in the config file + the same outcome when starting an app
![image](https://user-images.githubusercontent.com/126724797/223860906-75d75674-ac95-4c2e-a8cd-ac04c51dd340.png)
![image](https://user-images.githubusercontent.com/126724797/223861062-a260a59c-2baa-46b4-a4c1-0459f177ce0f.png)

**Reply by GrimCultZero** [reply]
UPD! Working now, thank you! @Koseng maybe you can help with the question: how can I configure the controls here? (e.g. set one of the knobs to pedestal lights and the slider to rudder trim)? I will be very grateful!

**Reply by Koseng** [reply]
Doc:
https://dev-tty.nl/X-Touch-Mini-FS2020/
https://dev-tty.nl/X-Touch-Mini-FS2020/configuration/configuration/
https://github.com/maartentamboer/X-Touch-Mini-FS2020/discussions/102

You should also have download mobiflight and installed the mobiflight wasm:
https://www.mobiflight.com/en/index.html
Mobiflight variables for the planes:
https://hubhop.mobiflight.com/presets/

Microsoft variables:
https://docs.flightsimulator.com/html/Programming_Tools/SimVars/Simulation_Variables.htm

**Reply by GrimCultZero** [reply]
Thank you so much! May the power be with you! Greetings from Ukraine!
